### DISCUSSION TITLE
HTTP -> HTTPS redirect

### DISCUSSION THREAD

**Discussion by Marvin0098** [distribution/distribution#4336] [Q&A] [answered]
Hello!

I have the docker registry running  and I can access it via HTTPS. However I have one application which tries to access the v2 endpoint via HTTP. Unfortunately I can't set the protocol for the app.

When I try to access the endpoint using http://registryip:5000/v2

I get the error: 

unexpected status code 400 Bad Request: Client sent an HTTP request to an HTTPS server.\n"

Is it possible to forward HTTP requests to HTTPS? I don't use nginx as an proxy, just using the docker registry directly.

**Comment by milosgajdos** [comment, marked as answer]
There is no way around this: either put a proxy in front of registry that will redirect HTTP -> HTTPS or make the application connect to HTTPS.

**Reply by Marvin0098** [reply]
Thanks for the quick reply, gonna put a proxy in front
